### PR TITLE
Convert df to pyspark DataFrame if it is pandas or pandas-on-Spark before writing

### LINKED ISSUE #179: Enable pandas and pandas-on-Spark DataFrames for Databricks
### Describe the feature
When a dbt python model returns a Pandas DataFrame, materialize it as a table in Databricks.

Currently, an error will be raised for Pandas DataFrames (whereas PySpark DataFrames work fine).

Same as https://github.com/dbt-labs/dbt-bigquery/issues/312 and https://github.com/dbt-labs/dbt-spark/issues/468

### Are you interested in contributing this feature?
I will submit a pull request for this 👍

### PR DESCRIPTION
**Pull request by dbeatty10**
resolves #179

### Description

Per https://github.com/databricks/dbt-databricks/pull/180#discussion_r974791617 removing `py_write_table` macro since https://github.com/dbt-labs/dbt-spark/pull/469 is merged.

### Checklist

- [X] I have run this code in development and it appears to resolve the stated issue
- [X] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have updated the `CHANGELOG.md` and added information about my change to the "dbt-databricks next" section.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 ### Features
 - Support python model through run command API, currently supported materializations are table and incremental. ([dbt-labs/dbt-spark#377](https://github.com/dbt-labs/dbt-spark/pull/377), [#126](https://github.com/databricks/dbt-databricks/pull/126))
+- Enable Pandas and Pandas-on-Spark DataFrames for dbt python models ([#181](https://github.com/databricks/dbt-databricks/pull/181))
```
Comment (ueshin) [inline CHANGELOG.md:5]: Could you also add the PR link in `dbt-spark`, like `[dbt-labs/dbt-spark#469](https://github.com/dbt-labs/dbt-spark/pull/469)`. See the above item.

**Review by ueshin — APPROVED** [review]
Otherwise, LGTM.

File: CHANGELOG.md (hunk 1 above)
Comment (dbeatty10) [reply CHANGELOG.md]: Good call. Just pushed that commit.

**Comment by ueshin** [comment]
Thanks for working on this! merging.

**Pull request merged**